### ISSUE TITLE
Unable to register multiple eloquent.creating listeners from within two traits

### ISSUE THREAD

**Issue by denjaland** [laravel/framework#32159]
- Laravel Version: 7.2.0
- PHP Version: 7.4.2
- Database Driver & Version: not applicable

### Description:
When having two traits that, in their `boot` method start listening for the `eloquent.creating` event, the second is never being ran.  We should be able to have more than one handler listening to the same event.

```php
trait TraitOne {
  public static function bootTraitOne() {
    self::creating(fn($model) => $model->trait_one = 'set');  
  }
}

trait TraitTwo {
  public static function bootTraitTwo() {
    self::creating(fn($model) => $model->trait_two = 'set');    
  }
}                   

class TestModel extends Illuminate\Database\Eloquent\Model {
  use TraitOne, TraitTwo;  
}

$tm = new testModel();
$tm->save();
```

Expected result is that both attributes `trait_one` and `trait_two` have been set, but they are not.  Only trait_one has a value, and trait_two does not.  When you only use TraitTwo, you'll notice it does have a value for trait_two, so the trait in itself is correct.

### Steps To Reproduce:
I have a created a tinkerwell snippet for this at https://web.tinkerwell.app/#/snippets/db02bbdd-77df-410f-9200-6b10514e50d0

Running that code, will obviously crash since there is no database, but the exception will show the query that Laravel is trying to execute...

![image](https://user-images.githubusercontent.com/12797576/77850923-122b7780-71d6-11ea-9391-6f0d7d31f3a2.png)

As you can tell from the error, the `trait_two` property has not been set.

**Comment by denjaland** [comment]
I just updated the tinkerell snippet to demonstrate that it's only the firstly registered trait that is being correctly executed.

Also added migration so you don't get the database error anymore, but the actual model as they are being persisted.

Check snippet at https://web.tinkerwell.app/#/snippets/f3ef114a-54e6-49a0-b450-b18a1504126d

**Comment by driesvints** [comment]
This isn't possible atm. Please open an feature request in the idea repo.

**Comment by denjaland** [comment]
Hi @driesvints,

I think you're a bit fast in closing this issue to be honest.
I have done some further investigation, and it seems the issue is that it _does_ work when you use regular functions, but it does not work when using new php 7.4 arrow functions syntax.

I have created a new snippet at https://web.tinkerwell.app/#/snippets/b2013f21-f7ca-48d6-befa-3192a64579c4 to desmonstrate the difference.

My guess is that - since arrow functions also return a value, the only difference that I can imagine - this is causing the strange behaviour.

I'm also not sure why you say it wouldn't be possible atm.  the static::creating() surely needs to support registering multiple listeners for the eloquent events?  If that isn't supported, you can't add different traits from different packages to the same models, and that would be pretty weird to be honest.

So in summary, I think that it _is_ supported, but that the new php 7.4 arrow function syntax is causing the disturbing behaviour.

Could you perhaps have a new look into it and possible reopen the issue?
I can workaround it now by converting the arrow functions to "regular" functions for now.

thanks, and have a nice day!
David

**Comment by driesvints** [comment]
@denjaland heya you're correct. I was mistaken. I'm not sure why the new syntax is causing issues here. 

Does it work when you define the methods in the model directly and use the new short arrow syntax?

**Comment by denjaland** [comment]
Hi @driesvints - thanks for reopening.
I haven't debugged, but just tried to follow the code, and believe https://github.com/laravel/framework/blob/b0609f429193d7a856816f7cdb8bdc81199c7004/src/Illuminate/Events/Dispatcher.php#L223 might be causing the issue.

An arrow function by default returns a value, so this is the different behaviour for "regular" functions.

**Comment by driesvints** [comment]
Then I believe there's not much we can do here sorry. You'll have to use regular functions.

**Comment by denjaland** [comment]
@driesvints I agree - I will create a pull request though for making sure that is documented ;-)

**Comment by denjaland** [comment]
@driesvints sorry to bother you again.  The docs (https://github.com/laravel/docs/blob/7.x/events.md#stopping-the-propagation-of-an-event) already explicitly refer that they would only stop propagation to other events when returning `false` from the handler.

If implementation is altered so that in stead of stopping propagation when `!is_null($response)`we only stop propagation when $response === false, implementation would be inline with the docs again.  It would also make more sense to me.

I'm sure in the future many developers will run into this issue when they start using the new arrow functions syntax.

What do you think?

**Comment by driesvints** [comment]
@denjaland no bother at all. I think the part you're reading there refers to https://github.com/laravel/framework/blob/b0609f429193d7a856816f7cdb8bdc81199c7004/src/Illuminate/Events/Dispatcher.php#L230 which is a different usecase.

**Comment by denjaland** [comment]
@driesvints You're right.
I can't find docs on event halting though and how I could disable this?  It seems it's enabled by default?

**Comment by driesvints** [comment]
@denjaland could be that's not documented yet. Maybe it's just for internal purposes.
